### DISCUSSION TITLE
routes REST API pass multiple `route_type`

### DISCUSSION THREAD

**Discussion by farfromrefug** [transitland/transitland#406] [Q&A] [answered]
I would like to filter `route_type` in the routes API. Especially my issue is that at some location i get weird routes (route_type=700) like in this https://transit.land/api/v2/rest/routes.json?lat=45.189885954254876&lon=5.724483132362368&radius=50&include_alerts
Having the possibility to pass multiple `route_type` could help (like `route_type:0,1,2,3,4,5,6,7,11,12` as per GTFS specs).
Maybe it would even make the query faster?

**Comment by drewda** [comment, marked as answer]
Thanks for this question and suggestion.

In general, the v2 REST API aims to only allow one possible value per query parameter. This helps to ensure that the database queries are performant, especially when multiple tables are joined behind the scenes and/or multiple query parameters are used within the query. So unfortunately we will not be able to provide the flexibility to query for multiple route types within a single request to the REST API routes endpoint. You may need to issue multiple queries to accomplish your goal, or have some filtering logic within your own code that issues the requests.

The Transitland v2 GraphQL API does provide means for including multiple queries, each with a different `route_type`, within a single overall API request.

**Reply by farfromrefug** [reply]
@drewda thanks for the answer. Quite sad that we have to make multiple requests. Will.make the user experience pretty slow. But won't it make your server slower to force users to make queries which ask for more data from your database? This what I have to do right now, then I filter the result afterwards.I understand the join question but if I don't pass a route_type the server will effectively do the same as if I was passing all route types as an array ? My point was about filtering to actually ask for less of the server but also to have lighter requests responses which is better for everyone I think.
As for the graphsql, could be a solution but I am developing opensource applications and sadly does not have to funds for the pro tier. 
Thanks for your answer
